sklearn/preprocessing.py

```diff
@@ -399,6 +404,120 @@ def __init__(self, copy=True, with_mean=True, with_std=True):
         super(Scaler, self).__init__(copy, with_mean, with_std)
 
 
+class RankScaler(BaseEstimator, TransformerMixin):
+    """Rank-standardize features to a percentile, in the range [0, 1].
+
+    Rank-scaling happens independently on each feature, by determining
+    the percentile of the feature value.
+    A feature value that is smaller than observed during fitting
+    will scale to 0.
+    A feature value that is larger than observed during fitting
+    will scale to 1.
+    A feature value that is the median will scale to 0.5.
+
+    Standardization of a dataset is a common requirement for many
+    machine learning estimators. Rank-scaling is useful when
+    estimators perform badly on StandardScalar features. Rank-scaling
+    is more robust than StandardScaler, because outliers can't have
+    large values post scaling. It is an empirical question whether
+    you want outliers to be given high importance (StandardScaler)
+    or not (RankScaler).
+
+    Parameters
+    ----------
+    n_ranks : int, 1000 by default
+        The number of different ranks possible.
+        i.e. The number of indices in the compressed ranking matrix
+        `sort_X_`.
+        This is an approximation, to save memory and transform
+        computation time.
+        e.g. if 1000, transformed values will have resolution 0.001.
+        If `None`, we store the full size matrix, comparable
+        in size to the initial fit `X`.
+
+    Attributes
+    ----------
+    `sort_X_` : array of ints, shape (n_samples, n_features)
```

Comment:
Actually we should just store an attribute named `quantiles_` with shape `(n_quantiles, n_features)`  as suggested by @jnothman .